src/librustc/ty/maps.rs

```diff
@@ -580,10 +598,17 @@ macro_rules! define_maps {
                        key,
                        span);
 
+                profq_msg!(tcx,
+                    ProfileQueriesMsg::QueryBegin(span.clone(),
+                                                  QueryMsg::$name(format!("{:?}", key))));
```

Comment:
The time, formatting in the compiler is not optimized for speed but rather code size (to avoid bloat from the presence of every `debug!` etc.).